src/main/java/seedu/address/model/task/Task.java

```diff
@@ -37,4 +79,59 @@ public void markTask() {
     public void unmarkTask() {
         isDone = false;
     }
+
+    /**
+     * Returns true if task contains any of the description or deadline keywords.
+     * By default, empty lists will return true.
+     *
+     * @param descriptionKeywords Possibly empty list containing keywords for {@code Description}.
+     * @param deadlineKeywords Possibly empty list containing keywords for {@code Deadline}.
+     * @return boolean indicating if task contains supplied keywords.
+     */
+    public boolean containsKeywords(List<Description> descriptionKeywords, List<Deadline> deadlineKeywords) {
+        return (descriptionKeywords.isEmpty() || descriptionKeywords.contains(description))
+                && (deadlineKeywords.isEmpty() || deadlineKeywords.contains(deadline));
+    }
+
+    /**
+     * Returns true if both persons have the same identity and data fields.
+     * This defines a stronger notion of equality between two persons.
+     */
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof Task)) {
+            return false;
+        }
+
+        Task otherTask = (Task) other;
+        return otherTask.getDescription().equals(getDescription())
+                && otherTask.getDeadline().equals(getDeadline())
+                && otherTask.getStatus().equals(getStatus());
```

Comment:
Maybe getStatus shouldn't be part of the equality check? As this would mean you can have two tasks with the same description and deadline if one is done and one is not done

Reply:
Following ab3 original implementation, when checking if person is the "same" using our own definition of equals (e.g. same name) the Person::isSamePerson is used. Person::equals is to check for whether objects are "same" as in everything the same. In actual use prob no need for distinction but i think it's used in testing

Tasks could follow a similar idea, will add a Task::isSameTask